include/alpaka/acc/AccGpuCudaRt.hpp

```diff
@@ -11,79 +11,13 @@
 
 #ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
 
-// Base classes.
 #    include <alpaka/acc/AccGpuUniformCudaHipRt.hpp>
-
-// Specialized traits.
-#    include <alpaka/acc/Traits.hpp>
-
-// Implementation details.
-#    include <alpaka/core/ClipCast.hpp>
-#    include <alpaka/core/Concepts.hpp>
-#    include <alpaka/core/Cuda.hpp>
-#    include <alpaka/dev/DevUniformCudaHipRt.hpp>
-
-#    include <typeinfo>
+#    include <alpaka/core/ApiCudaRt.hpp>
 
 namespace alpaka
 {
-    template<typename TAcc, typename TDim, typename TIdx, typename TKernelFnObj, typename... TArgs>
-    class TaskKernelGpuUniformCudaHipRt;
-
-    //! The GPU CUDA accelerator.
-    //!
-    //! This accelerator allows parallel kernel execution on devices supporting CUDA.
     template<typename TDim, typename TIdx>
-    class AccGpuCudaRt final
-        : public AccGpuUniformCudaHipRt<TDim, TIdx>
-        , public concepts::Implements<ConceptUniformCudaHip, AccGpuUniformCudaHipRt<TDim, TIdx>>
-    {
-        static_assert(
-            sizeof(TIdx) >= sizeof(int),
-            "Index type is not supported, consider using int or a larger type.");
-
-    public:
-        ALPAKA_FN_HOST_ACC AccGpuCudaRt(Vec<TDim, TIdx> const& threadElemExtent)
-            : AccGpuUniformCudaHipRt<TDim, TIdx>(threadElemExtent)
-        {
-        }
-    };
-
-    namespace trait
-    {
-        //! The GPU CUDA accelerator accelerator type trait specialization.
-        template<typename TDim, typename TIdx>
-        struct AccType<AccGpuCudaRt<TDim, TIdx>>
-        {
-            using type = AccGpuCudaRt<TDim, TIdx>;
-        };
-
-        //! The GPU CUDA accelerator name trait specialization.
-        template<typename TDim, typename TIdx>
-        struct GetAccName<AccGpuCudaRt<TDim, TIdx>>
-        {
-            ALPAKA_FN_HOST static auto getAccName() -> std::string
-            {
-                return "AccGpuCudaRt<" + std::to_string(TDim::value) + "," + typeid(TIdx).name() + ">";
-            }
-        };
-
-        //! The GPU CUDA accelerator execution task type trait specialization.
-        template<typename TDim, typename TIdx, typename TWorkDiv, typename TKernelFnObj, typename... TArgs>
-        struct CreateTaskKernel<AccGpuCudaRt<TDim, TIdx>, TWorkDiv, TKernelFnObj, TArgs...>
-        {
-            ALPAKA_FN_HOST static auto createTaskKernel(
-                TWorkDiv const& workDiv,
-                TKernelFnObj const& kernelFnObj,
-                TArgs&&... args)
-            {
-                return TaskKernelGpuUniformCudaHipRt<AccGpuCudaRt<TDim, TIdx>, TDim, TIdx, TKernelFnObj, TArgs...>(
-                    workDiv,
-                    kernelFnObj,
-                    std::forward<TArgs>(args)...);
-            }
-        };
-    } // namespace trait
-} // namespace alpaka
+    using AccGpuCudaRt = AccGpuUniformCudaHipRt<ApiCudaRt, TDim, TIdx>;
```

Comment:
We started to discuss this line [here](https://github.com/fwyzard/alpaka/commit/34f251aec7be8565a081aecc872467d92e0182af#r69328577) already.

summery: removing the struct would give the wrong naming via `GetAccName` and type via `AccType`

Reply:
@fwyzard I suggest to keep the implementation with the struct instead of the alias to keep the separation of the HIP and CUDA accelerator.

Reply:
> removing the struct would give the wrong naming via `GetAccName` and type via `AccType`

Actually it doesn't, because `GetAccName` returns a name based on `API::name`:
```c++
        template<typename API, typename TDim, typename TIdx>
        struct GetAccName<AccGpuUniformCudaHipRt<API, TDim, TIdx>>
        {
            ALPAKA_FN_HOST static auto getAccName() -> std::string
            {
                return "AccGpu" + API::name + "Rt<" + std::to_string(TDim::value) + "," + typeid(TIdx).name() + ">";
            }
        };
 ```

And `AccType` should be `AccGpuUniformCudaHipRt<API, ...>` which is the same type as either `AccGpuCudaRt<...>` or `AccGpuHipRt<...>`.